apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -5,59 +5,420 @@
+        kwargs: Dict[str, object] = {"result_order": self.sr.result_order}
+        for coord_chunk in _coords_strider(
+            self.coords[self.major_axis],
+            self.sr.shape[self.major_axis],
+            self.size[0],
+        ):
+            self.sr.reset(**kwargs)
+            step_coords = [*self.coords]
+            step_coords[self.major_axis] = coord_chunk
+            self.array._set_reader_coords(self.sr, step_coords)
+            yield pa.concat_tables(_arrow_table_reader(self.sr)), tuple(
+                self.joinids[0 : self.major_axis]
+                + [pa.array(coord_chunk)]
+                + self.joinids[self.major_axis + 1 : self.ndim]
+            )
+
+    def _reindexed_table_reader(
+        self, _pool: Optional[ThreadPoolExecutor] = None
+    ) -> BlockwiseSingleAxisTableIter:
+        """Private. Blockwise table reader w/ reindexing. Helper function for sub-class use"""
+        for tbl, coords in self._if_eager(self._table_reader(), _pool):
+            pytbl = {}
+            for d in range(self.ndim):
+                col = tbl.column(f"soma_dim_{d}")
+                if d in self.axes_to_reindex:
+                    if d == self.major_axis:
+                        col = pd.Index(coords[self.major_axis]).get_indexer(col.to_numpy())  # type: ignore[no-untyped-call]
+                    else:
+                        col = self.minor_axes_indexer[d].get_indexer(col.to_numpy())  # type: ignore[no-untyped-call]
+                pytbl[f"soma_dim_{d}"] = col
+            pytbl["soma_data"] = tbl.column("soma_data")
+            yield pa.Table.from_pydict(pytbl), coords
+
+
+class BlockwiseTableReadIter(BlockwiseReadIterBase[BlockwiseTableReadIterResult]):
+    """Blockwise iterator over `Arrow Table <https://arrow.apache.org/docs/python/generated/pyarrow.Table.html>`_ elements"""
+
+    def _create_reader(self) -> Iterator[BlockwiseTableReadIterResult]:
+        """Private. Blockwise Arrow Table iterator, restricted to a single axis"""
+        if self.eager and not self.pool:
+            with ThreadPoolExecutor() as _pool:
+                yield from self._reindexed_table_reader(
+                    _pool
+                ) if self.axes_to_reindex else self._table_reader()
+        else:
+            yield from self._reindexed_table_reader(
+                _pool=self.pool
+            ) if self.axes_to_reindex else self._table_reader()
+
+
+class BlockwiseScipyReadIter(BlockwiseReadIterBase[BlockwiseScipyReadIterResult]):
+    """Blockwise iterator over `SciPy sparse matrix <https://docs.scipy.org/doc/scipy/reference/sparse.html>`_ elements"""
+
+    def __init__(
+        self,
+        array: "SparseNDArray",
+        sr: clib.SOMAArray,
+        coords: options.SparseNDCoords,
+        axis: Union[int, Sequence[int]],
+        *,
+        size: Optional[Union[int, Sequence[int]]] = None,
+        reindex_disable: Optional[Union[int, Sequence[int]]] = None,
```

Comment:
I'm not a *hard* no, but I worry that doing so would overload the meaning of this parameter a bit.

Reply:
nice idea, will add.  more sugar, more unit tests :-)

Reply:
I'm ambivalent - but have it implemented.  Who wants to decide?

Reply:
I suggested it because while test driving the API, I mistakenly assumed it was a bool flag. :) After RTFM, it made sense, but it seemed like a minor trap. If we want to keep typing as is, maybe consider renaming the param to something, e.g. `reindex_disable_on_axis`

Reply:
@atolopko-czi's `reindex_disable_on_axis` suggestion seems like good middle ground. The arg name is a bit verbose but its intent is clear.

Reply:
I see disabling re-indexing a bit more of an advanced functionality, and will also likely happen on one axis frequently. Let's not add more cognitive dissonance with `bool` option. Re-naming the parameter is a +1. 

Thus

- `reindex_disable_on_axis` should be the name
- `reindex_disable_on_axis` should not take bool.

This is one of these cases where we can easily add the sugar in the future if needed, whereas removing it in the future can get complicated.